### PR TITLE
Created Unit Test for fun refreshIfRequired

### PR DESCRIPTION
**Pull request by Haz3-jolt**
## Purpose / Description
Added Unit test to function with @NeedsTest.

## Fixes
* Adds test for fun refreshIfRequired, which was tagged with @NeedsTest("if opExecuted is called while activity is in the background, audio plays onResume").

## Approach
- Making refreshIfRequired public so it is available for testing.
- Create a companion object for testing.
- Unit test with multiple cases to test for edge cases.
## How Has This Been Tested?
Ran the Unit test multiple times, and ensured it is passing.

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 1]
```diff
@@ -281,7 +281,7 @@ abstract class AbstractFlashcardViewer :
      * @see opExecuted
      * @see refreshIfRequired
      */
-    private var refreshRequired: ViewerRefresh? = null
+    var refreshRequired: ViewerRefresh? = null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:284]: `@VisibleForTesting`

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 2]
```diff
@@ -574,8 +574,7 @@ abstract class AbstractFlashcardViewer :
      *
      * If the activity is NOT [RESUMED], wait until [onResume]
      */
-    @NeedsTest("if opExecuted is called while activity is in the background, audio plays onResume")
-    private fun refreshIfRequired(isResuming: Boolean = false) {
+    fun refreshIfRequired(isResuming: Boolean = false) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:577]: ditto

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/ViewerRefresh.kt [hunk 3]
```diff
@@ -21,6 +21,9 @@ import anki.collection.OpChanges
 /** Data for a deferred refresh of the CardViewer */
 data class ViewerRefresh(val queues: Boolean, val note: Boolean, val card: Boolean) {
     companion object {
+        val UpdateCard: ViewerRefresh
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/ViewerRefresh.kt:24]:
```suggestion
        val updateCard: ViewerRefresh
```

Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/ViewerRefresh.kt:24]: If this is only used for testing, move it into the test class

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 4]
```diff
@@ -96,6 +100,36 @@ class AbstractFlashcardViewerCommandTest : RobolectricTest() {
         assertThat(viewer.lastFlag, equalTo(Flag.ORANGE.code))
     }
 
+    @Test
+    fun testRefreshIfRequired() {
+        // Create an ActivityController instance for AbstractFlashcardViewer
+        val controller = Robolectric.buildActivity(AbstractFlashcardViewerTest.NonAbstractFlashcardViewer::class.java)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:106]: this needs `.use { }` to ensure it's cleaned up

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 5]
```diff
@@ -96,6 +100,36 @@ class AbstractFlashcardViewerCommandTest : RobolectricTest() {
...
+
+        // Case 1: Activity is resuming, refreshRequired is not null
+        var viewer = controller.create().start().get()
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:109]: I'd consider `with (controller.get()` rather than using a `var`

**Review by david-allison — CHANGES_REQUESTED** [review]
Quality is great! 

You don't need to make all the changes here, just comment and resolve the ones you don't like

**Comment by david-allison** [comment]
```
> Task :AnkiDroid:ktlintMainSourceSetCheck FAILED
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/ViewerRefresh.kt:25:1 Unexpected indentation (20) (should be 12)

```

**Comment by david-allison** [comment]
> [-] You have a descriptive commit message with a short title (first line, max 50 chars).

Use `[x]` and `[ ]` and you get:

> [x] You have a descriptive commit message with a short title (first line, max 50 chars).
> [ ] You have a descriptive commit message with a short title (first line, max 50 chars).

**Review by david-allison — APPROVED** [review]
Looks great! All comments are optional

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 6]
```diff
@@ -574,8 +575,8 @@ abstract class AbstractFlashcardViewer :
      *
      * If the activity is NOT [RESUMED], wait until [onResume]
      */
-    @NeedsTest("if opExecuted is called while activity is in the background, audio plays onResume")
-    private fun refreshIfRequired(isResuming: Boolean = false) {
+    @VisibleForTesting
+    internal fun refreshIfRequired(isResuming: Boolean = false) {
```
[the remark this answers is not in the document]
Comment (Haz3-jolt) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:579]: I tried with private, but it's invisible to the test even with the tag, so I used internal. Is there another way? Or should I just make it public.

[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:579]: internal has little meaning for us, and it's more characters. Public is fine

**Review by lukstbit — APPROVED** [review]
Thanks!
